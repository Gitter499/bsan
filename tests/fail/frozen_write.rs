//@run: 1
fn main() {
    let x = 1;
    let rx = &x;
    foo(rx as *const i32);
}

fn foo(ptr: *const i32) {
    unsafe { *(ptr as *mut i32) = 1 };
}