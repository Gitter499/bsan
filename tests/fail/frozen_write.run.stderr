An error occurred: UndefinedBehavior(AliasingViolation(TreeError { error_kind: ChildAccessForbidden(Permission { inner: Frozen }), alloc_id: ALLOC, error_offset: 0, conflicting_info: NodeDebugInfo { tag: <TAG>, name: None, history: History { tag: <TAG>, created: (Span(0), Permission { inner: Frozen }), events: [] } }, access_cause: Explicit(Write), accessed_info: NodeDebugInfo { tag: <TAG>, name: None, history: History { tag: <TAG>, created: (Span(0), Permission { inner: Frozen }), events: [] } } }))


